### PR TITLE
As pre-3.5 support is dropped, remove support in the code too.

### PR DESCRIPTION
**Pull request by svaningelgem**
Work in progress. But maybe you can explain 1 thing. What is the cast_to_int/long/short/... supposed to do?

### CODE REVIEW

File: docs/sphinx/conf.py [hunk 1]
```diff
@@ -22,12 +22,12 @@
 # If extensions (or modules to document with autodoc) are in another directory,
 # add these directories to sys.path here. If the directory is relative to the
 # documentation root, use os.path.abspath to make it absolute, like shown here.
-#sys.path.insert(0, os.path.abspath('.'))
```
Comment (svenkreiss) [inline docs/sphinx/conf.py:25]: This templates sphinx config file has meaningful (not)spaces per line. Please revert.

File: pysparkling/context.py [hunk 2]
```diff
@@ -21,12 +21,8 @@
 from .rdd import RDD, EmptyRDD
 from .task_context import TaskContext
 
-# Python 2 compatibility
 if not hasattr(time, 'perf_counter'):
```
Comment (svenkreiss) [inline pysparkling/context.py:24]: can this be removed completely now?

File: pysparkling/fileio/textfile.py [hunk 3]
```diff
@@ -9,11 +9,7 @@
 
 log = logging.getLogger(__name__)
 
-# there is no basestring in Python 3, so define string_types:
-try:
-    string_types = basestring
-except NameError:
-    string_types = str
+string_types = str
```
Comment (svenkreiss) [inline pysparkling/fileio/textfile.py:12]: can this be removed now? Is it just `str` everywhere?

File: pysparkling/sql/types.py [hunk 4]
```diff
@@ -1159,20 +1133,20 @@ def convert_dict(obj):
...
+    IntegerType: (int,),
+    LongType: (int,),
     FloatType: (float,),
     DoubleType: (float,),
     DecimalType: (decimal.Decimal,),
-    StringType: (str, unicode),
+    StringType: (str,),
     BinaryType: (bytearray,),
-    DateType: (datetime.date, datetime.datetime),
+    DateType: (datetime.date, datetime.datetime,),
```
Comment (svenkreiss) [inline pysparkling/sql/types.py:1145]: can you revert adding commas to tuples with more than one element?

**Review by svenkreiss — CHANGES_REQUESTED** [review]
Thanks! I have added some questions/requests.

File: pysparkling/sql/types.py (hunk 4 above)
Comment (svaningelgem) [reply pysparkling/sql/types.py:1145]: Ok, that's easy to do... I just think it is nicer that all are looking equal. I know it doesn't matter, but I like uniformity.

Comment (svenkreiss) [reply pysparkling/sql/types.py:1164]: I disagree.

**Review by svenkreiss — APPROVED** [review]
Thanks for going through all this!

**Pull request merged**